app/services/course/assessment/submission/zip_download_service.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+class Course::Assessment::Submission::ZipDownloadService
+  class << self
+    # Downloads the submissions and zip them.
+    #
+    # @param [CourseUser] course_user The course user downloading the submissions.
+    # @param [Course::Assessment] assessment The assessments to download submissions from.
+    # @param [String] limit The subset of submissions to download.
+    # @return [String] The path to the zip file.
+    def download_and_zip(course_user, assessment, limit)
+      service = new(course_user, assessment, limit)
+      ActsAsTenant.without_tenant do
+        service.send(:download_to_base_dir)
+      end
+      service.send(:zip_base_dir)
+    end
+  end
+
+  private
+
+  def initialize(course_user, assessment, limit)
+    @course_user = course_user
+    @assessment = assessment
+    @limit = limit
+    @base_dir = Dir.mktmpdir('coursemology-download-')
+  end
+
+  # Downloads each submission to its own folder in the base directory.
+  def download_to_base_dir
+    submissions = @assessment.submissions.by_users(student_ids)
+    submissions.each do |submission|
+      submission_dir = create_folder(@base_dir, submission.course_user.name)
+      download_answers(submission, submission_dir)
+    end
+  end
+
+  # Downloads each answer to its own folder in the submission directory.
+  def download_answers(submission, submission_dir)
+    answers = submission.answers.select { |a| a.specific.respond_to?(:download) }
```

Comment:
Have to be careful with this line, because it is dependent on the implementation of this feature, ie. if `download` isn't implemented, the answer is not downloadable. In the case where the `answer` model has a method `download` which raises an error, and requires the specific model to implement it, then this line would not be appropriate.